### DISCUSSION TITLE
The voice chat on self-hosted server

### DISCUSSION THREAD

**Discussion by BH2VSQ** [stoatchat/discussions#593] [Q&A]
I build my own server with the repo sel-hosted. When I switch to the voice channel and click the button, my broswer returned "VosoUnavailable". So whether I need to install other services for the voice channel or I just need to edit some config file to enable the voice chat?

**Comment by siddhantprateek** [comment]
I'm having the same, I am unable to join a voice channel on my self-hosted version. Although the Revolt API server logs show a successful response after attempting to join the call:

```bash
revolt-api-1  |  INFO  rocket::server                                                > POST /channels/01J8S371F6EWFG6Z6TT26KBJYS/join_call:
revolt-api-1  |  INFO  _                                                             > Matched: (call) POST /channels/<target>/join_call
revolt-api-1  |  INFO  _                                                             > Outcome: Success
revolt-api-1  |  INFO  _                                                             > Response succeeded.
```

In the browser's developer tools, the network tab shows a response indicating an error:
```json
{
    "type": "VosoUnavailable",
    "location": "crates/delta/src/routes/channels/voice_join.rs:29:20"
}
```

- Is the voice-chat feature available in self-hosted version?

**Comment by kapralnsk** [comment]
Hey guys, have you been able to resolve this? Just set up my instance and attempt to join a voice channel results in the same `VosoUnavailable` code. No idea how to enable voice on my instance :(

If it helps, I'm running revolt on Ubuntu 22.04, via docker compose. It's hosted on a droplet in Digital Ocean, I do have my own domain set up for it, and apart from voice basic stuff seems to work (I was able to register w/ invite only turned on and send messages to text channels). Happy to provide more info if needed

**Reply by siddhantprateek** [reply]
Based on my conversation with the Revolt Support team last month, we need a voice server https://github.com/revoltchat/vortex to enable the voice feature. Currently, it's on hold.

**Reply by parzifox** [reply]
in the version of the beginning of the year, I managed to launch vortex. but it's not working right now. More precisely, vortex itself starts but revolt does not connect to it

**Reply by gru2007** [reply]
> in the version of the beginning of the year, I managed to launch vortex. but it's not working right now. More precisely, vortex itself starts but revolt does not connect to it

I have installed it successfully - check it out here https://discord.stormgalaxy.com

**Comment by parzifox** [comment]
[Vortex](https://github.com/revoltchat/vortex)

vortex .env
```
WS_URL=https://voso.test.com
VOSO_WS_HOST=wss://voso.test.com
HTTP_HOST=0.0.0.0:8080
RTC_IPS=0.0.0.0,YOU SERVER IP
RTC_MIN_PORT=10001
RTC_MAX_PORT=11000
VOSO_MANAGE_TOKEN=moi84
MANAGE_TOKEN=moi84
DISABLE_RTP=0
```

That's all I know at the moment.

But at the moment I don't know how to make revolt work in the current version with vortex.

**Comment by johnsnowdies** [comment]
So, is there a way to run self-hosted revolt with voice chat?

If somebody can do it, keep this discussion up to date please

**Reply by parzifox** [reply]
At the moment, I only know about the vortex, but I don't know how to connect them in this version, at the beginning of the year I managed to get it all started.

**Reply by johnsnowdies** [reply]
There is a link to "[new branch](https://github.com/revoltchat/vortex/tree/vortex)" but this branch was updated 2 years ago and looks outdated

I will try to build and run vortex and keep you updated

**Reply by gru2007** [reply]
Yeap it works, it just needs correct Proxy for SSL and WebSocket

**Reply by jaguarn** [reply]
> Yeap it works, it just needs correct Proxy for SSL and WebSocket

Hey! Do you mean proxy for ssl on vortex? Or? I have running self-hosted, and i have running vortex in same docker. Amd vortex is available, but in connect to voice chat it's still 400. :\

**Comment by parzifox** [comment]
I have successfully launched a voice chat!!!

Revolt.toml
```
voso_legacy = "https://voso.test.com"
voso_legacy_ws = "wss://voso.test.com"
```
and search tocken place 
`voso_legacy_token = "moi84"`

.env in vortex

```
WS_URL=wss://voso.test.com
HTTP_HOST=0.0.0.0:8080
RTC_IPS=0.0.0.0,HOSTING IP
RTC_MIN_PORT=10001
RTC_MAX_PORT=11000
VOSO_MANAGE_TOKEN=moi84
MANAGE_TOKEN=moi84
DISABLE_RTP=0
```
and need open ports 10001-11000
```
RTC_MIN_PORT=10001
RTC_MAX_PORT=11000
```

**Reply by parzifox** [reply]
Stability leaves much to be desired. Periodically, the voice is turned off..

**Reply by BasiliyWolf** [reply]
Tell me how you started it?
Did you change anything in Docker or somewhere else?
I'm trying, but the service starts and Revolt still doesn't see it

**Reply by johnsnowdies** [reply]
I keep self-hosted revolt behind caddy server with default configuration with custom domain (https) only, following all steps from the [readme](https://github.com/revoltchat/self-hosted/blob/master/README.md). 

[Vortex ](https://github.com/revoltchat/vortex) installed on another server to keep it proxy_passed via nginx.

Used the configuration [parzifox](https://github.com/parzifox) made, it works but voice is unstable (it lost connection randomly)

**Reply by parzifox** [reply]
I'm trying to find a disynchronization problem.

At the moment, there is only one clue.
`WARN  revolt_bonfire::websocket > Error while reading an event from 172.20.0.2:41162: Protocol(ResetWithoutClosingHandshake)`

As far as I understand, at some point the handshake exchange breaks down

**Reply by Aspicson** [reply]
> I keep self-hosted revolt behind caddy server with default configuration with custom domain (https) only, following all steps from the [readme](https://github.com/revoltchat/self-hosted/blob/master/README.md).
> 
> [Vortex ](https://github.com/revoltchat/vortex) installed on another server to keep it proxy_passed via nginx.
> 
> Used the configuration [parzifox](https://github.com/parzifox) made, it works but voice is unstable (it lost connection randomly)

How did you manage to build Vortex? Do you have an example of a Dockerfile, or did you successfully build it through Windows WSL? I'm encountering countless errors related to missing libraries, and when I try to update them to the latest versions, even more errors pop up. I can't get Vortex to run and really need some help. I've tried both using the official Dockerfile from the master branch and running cargo build, but I'm still stuck.

**Reply by JoCat** [reply]
What version did you use? I was able to build what is currently in the master, but for some reason it is not listening to the port I specified. And the version from tag 0.3.0-alpha.1 is not built because of an error.

**Reply by johnsnowdies** [reply]
> > I keep self-hosted revolt behind caddy server with default configuration with custom domain (https) only, following all steps from the [readme](https://github.com/revoltchat/self-hosted/blob/master/README.md).
> > [Vortex ](https://github.com/revoltchat/vortex) installed on another server to keep it proxy_passed via nginx.
> > Used the configuration [parzifox](https://github.com/parzifox) made, it works but voice is unstable (it lost connection randomly)
> 
> How did you manage to build Vortex? Do you have an example of a Dockerfile, or did you successfully build it through Windows WSL? I'm encountering countless errors related to missing libraries, and when I try to update them to the latest versions, even more errors pop up. I can't get Vortex to run and really need some help. I've tried both using the official Dockerfile from the master branch and running cargo build, but I'm still stuck.

I just copied default.nix from self-hosted repo, run nix-shell and cargo build && cargo run, clear ubuntu 20.04

**Reply by JoCat** [reply]
I found the reason, the voice channels don't work with the old frontend (the one that goes self-hosted), after replacing it with the new one it became possible to connect to the voice channels.

**Reply by jaguarn** [reply]
> I found the reason, the voice channels don't work with the old frontend (the one that goes self-hosted), after replacing it with the new one it became possible to connect to the voice channels.

Hey. Sorry to disturb. Is this approach working? With replacing frontend? How you made it work?)

**Reply by JoCat** [reply]
> Hey. Sorry to disturb. Is this approach working? With replacing frontend? How you made it work?)

Hi, no, that was my misjudgment. In a [neighboring comment](https://github.com/orgs/revoltchat/discussions/593#discussioncomment-10892829), I explained what I was doing

**Comment by kotazzz** [comment]
If someone also managed to laconically make voice chats work, please let us know. I would also like to repeat your success, preferably compatible with a self-hosted repository. I recently suffered from the fact that media was not loaded (even the avatar did not change), as a result, changing the port in candyfile helped. It wouldn't be bad if it was enough to change compose.yml in order to make voice channels work

**Reply by BasiliyWolf** [reply]
I am currently studying the source code to understand how to set it all up. It seems the problem is in the ports and that the main branch does not specify the right ports.

**Reply by BasiliyWolf** [reply]
For those who will compile on the latest version 20241002-7, then for WSS and personal nginx you need to specify port not 9000, but 14703

**Reply by johnsnowdies** [reply]
Btw autumn service port is 3000 not 14704

**Reply by JoCat** [reply]
I have everything running in this configuration: self-hosted with [port patching for autumn](https://github.com/revoltchat/self-hosted/pull/103), [vortex which I deployed to docker](https://hub.docker.com/r/revoltchat/vortex)  the one built in person doesn't work for some reason (not listening to ports), ~~and a separately running [new frontend](https://github.com/revoltchat/frontend) which is prescribed in caddy (I give the already built frontend pre-specified in the compose file)~~.

![image](https://github.com/user-attachments/assets/61a48d1d-2511-4265-bf89-069c694a1fea)
![image](https://github.com/user-attachments/assets/424fa63c-b347-4071-8053-b2b2e7505031)


I didn't check the quality of sound transmission because I have no one to check it with, all my friends are already asleep.
___

**UPD.** First it turned out that the frontend was not proxying, after which I reconfigured everything, but it turned out that now the cache was getting in the way. After clearing the cache completely, a new frontend appeared that won't let me into my account. Seems to be a difference in the api.
___

**UPD 2.** Rolled the frontend back, looks like it's definitely working now....
![image](https://github.com/user-attachments/assets/0f6ea532-5cfe-400b-abfb-7e63c97072eb)
![image](https://github.com/user-attachments/assets/b925cc95-5093-4f30-8caa-9db8ebfab056)

**Reply by parzifox** [reply]
> I have everything running in this configuration: self-hosted with [port patching for autumn](https://github.com/revoltchat/self-hosted/pull/103), [vortex which I deployed to docker](https://hub.docker.com/r/revoltchat/vortex) the one built in person doesn't work for some reason (not listening to ports), ~and a separately running [new frontend](https://github.com/revoltchat/frontend) which is prescribed in caddy (I give the already built frontend pre-specified in the compose file)~.
> 
> ![image](https://private-user-images.githubusercontent.com/13483132/375175536-61a48d1d-2511-4265-bf89-069c694a1fea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1NTEzNTQsIm5iZiI6MTcyODU1MTA1NCwicGF0aCI6Ii8xMzQ4MzEzMi8zNzUxNzU1MzYtNjFhNDhkMWQtMjUxMS00MjY1LWJmODktMDY5YzY5NGExZmVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEwVDA5MDQxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiNTEzMDkwMzc0OGFlY2MzODMwMGQ4MzczOTExYjliNmQ3OWM0MjEyMDJkYmM0MDgzMDYyOGRkNmI4YjFmNDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0nyR84nCw5EmzYpfIn1NHHbfdOp__j2dMIGtUQUk_6k) ![image](https://private-user-images.githubusercontent.com/13483132/375169783-424fa63c-b347-4071-8053-b2b2e7505031.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1NTEzNTQsIm5iZiI6MTcyODU1MTA1NCwicGF0aCI6Ii8xMzQ4MzEzMi8zNzUxNjk3ODMtNDI0ZmE2M2MtYjM0Ny00MDcxLTgwNTMtYjJiMmU3NTA1MDMxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEwVDA5MDQxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhNDQ5NDc5NjM4NWE3YzhlZmY3YTQ0MTlmYjZhNjNjNzM1MDY2ZDJlZmEwMzM0YjVmN2Q5MTM4M2M2OGZjNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.W_NtHEy5c2cc_eCZjkdlpzSMow-TaETWWwpgOj2ThHo)
> 
> I didn't check the quality of sound transmission because I have no one to check it with, all my friends are already asleep.
> 
> **UPD.** First it turned out that the frontend was not proxying, after which I reconfigured everything, but it turned out that now the cache was getting in the way. After clearing the cache completely, a new frontend appeared that won't let me into my account. Seems to be a difference in the api.
> 
> **UPD 2.** Rolled the frontend back, looks like it's definitely working now.... ![image](https://private-user-images.githubusercontent.com/13483132/375176959-0f6ea532-5cfe-400b-abfb-7e63c97072eb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1NTEzNTQsIm5iZiI6MTcyODU1MTA1NCwicGF0aCI6Ii8xMzQ4MzEzMi8zNzUxNzY5NTktMGY2ZWE1MzItNWNmZS00MDBiLWFiZmItN2U2M2M5NzA3MmViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEwVDA5MDQxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI3OTcwMWQ0MTVlNTEzZThjYmVmMjBiODEzOTgxZTA1YzRlNTQ3ODA2NGQ4MGU0YWQ4YjI5MDQxNmJmYTZlY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Z6k7PuDOf0qJWH_QAmtuodQ6FNZFQeOh2JfPEv9FyT8) ![image](https://private-user-images.githubusercontent.com/13483132/375176796-b925cc95-5093-4f30-8caa-9db8ebfab056.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1NTEzNTQsIm5iZiI6MTcyODU1MTA1NCwicGF0aCI6Ii8xMzQ4MzEzMi8zNzUxNzY3OTYtYjkyNWNjOTUtNTA5My00ZjMwLThjYWEtOWRiOGViZmFiMDU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEwVDA5MDQxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAzY2EyODhiMmYwNTYxZTY5NjgxZDFmODUyZDkxNDBmZGFkOWVjOTE2NzM0YTc4ZGJjMzIyMjdhMTczOGIxNWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7oSO4FCDfk85wOJv6NrWf0ke4fX-y8XmlkMJwpdjWc8)

Did you manage to increase the stability of the voice chat?

**Reply by JoCat** [reply]
> Did you manage to increase the stability of the voice chat?

No, just went in to check now, the voice channels have stopped working again. The join_call request returns a 400 error
![image](https://github.com/user-attachments/assets/e5725b1e-08eb-43cb-9e05-e05f45ac01b4)

**UPD** I'm an idiot, I deleted the domain in CloudFlare by mistake 🤣

**Reply by JoCat** [reply]
As a result, I and my friends were able to connect to the channel, but there was no sound, most likely because the sound settings were not saved

**Reply by johnsnowdies** [reply]
> Did you manage to increase the stability of the voice chat?

Are there any updates? I created a new [issue](https://github.com/revoltchat/vortex/issues/31) about the next Vortex release, but so far, there have been no updates. I will try debugging the old version today.

In my setup, the most critical bug right now is random disconnection from the voice chat without any state updates (users still see all voice chat members as connected).

Another important missing UI feature is the list of users connected to the voice channel, which should appear under the channel name.

I’ll focus on this bug and feature. If there are no updates on the issue I linked above, I will fork the frontend and Vortex repositories for experimental development purposes.

**Comment by KamiliaBlow** [comment]
If you can figure it out completely, it would be nice if you could provide full instructions on how to work with vortex, at the moment I'm stuck at compile time and have no idea how to solve the problem. 

```
Compiling mediasoup-sys v0.7.2
error: failed to run custom build command for `mediasoup-sys v0.7.2`
note: To improve backtraces for build dependencies, set the CARGO_PROFILE_DEV_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.

Caused by:
  process didn't exit successfully: `C:\Users\Администратор\vortex\target\debug\build\mediasoup-sys-64158e514649891f\build-script-build` (exit code: 101)
  --- stderr
  thread 'main' panicked at C:\Users\Администратор\.cargo\registry\src\index.crates.io-6f17d22bba15001f\mediasoup-sys-0.7.2\build.rs:131:10:
  Failed to start: Error { kind: NotFound, message: "program not found" }
  stack backtrace:
     0: std::panicking::begin_panic_handler
               at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library\std\src\panicking.rs:665
     1: core::panicking::panic_fmt
               at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library\core\src\panicking.rs:74
     2: core::result::unwrap_failed
               at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library\core\src\result.rs:1679
     3: std::rt::lang_start
     4: std::rt::lang_start
     5: std::rt::lang_start
     6: std::rt::lang_start
     7: std::rt::lang_start
     8: std::rt::lang_start_internal::closure$2
               at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library\std\src\rt.rs:141
     9: std::panicking::try::do_call
               at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library\std\src\panicking.rs:557
    10: std::panicking::try
               at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library\std\src\panicking.rs:521
    11: std::panic::catch_unwind
               at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library\std\src\panic.rs:350
    12: std::rt::lang_start_internal
               at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library\std\src\rt.rs:141
    13: std::rt::lang_start
    14: main
    15: invoke_main
               at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
    16: __scrt_common_main_seh
               at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
    17: BaseThreadInitThunk
    18: RtlUserThreadStart
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
  ```
  
  In my case, I'm working through WS 2022 and Docker Desktop

**Comment by jaguarn** [comment]
Guy, did anyone face this issue: 
![image](https://github.com/user-attachments/assets/76bd34fb-cbde-473c-969e-6862f4cdcf7b)
Revolt installed from the self-host, then i patched proxy (autumn to 3000). Added vortex to the compose with open 8080 port. Added pretty same .env to previos. And now - site/api/fe is working with ssl and vortex is loading separately, but only by http from outside. And on every join_call - i have this 400 error :\

**Comment by mr-each** [comment]
Hello there!
I'm looking for some kind of middle-ground conclusion on the topic of hosting Revolt with working VC. There are some closed git issues where people tried to make Vortex working with Revolt, and succeeded, but they are lacking instructions and a bit outdated.
I do not have enough knowledge to get things done so I am asking for help from big brains here. Maybe someone will be so generous to share a step-by-step guide with configs for others to blindly copy-paste...

I've managed to boot vortex (cloning [vortex](https://github.com/revoltchat/vortex) repo, adding [default.nix](https://github.com/revoltchat/backend/blob/main/default.nix) and building it like [johnsnowdies](https://github.com/orgs/revoltchat/discussions/593#discussioncomment-10897635) did) on a VM separate from Revolt. Both are set behind NPM. From this point VC became joinable, but there is no sound transferred as well as there are no errors neither in browser nor in Vortex console. I believe that I've screwed NPM configs coz I'm redirecting vortex.some.domain to http://vortex_host:8080 + websockets support w/o any additional advanced settings.

TL;DR
A couple of questions for those who ARE using Vortex at it's current state:
Is it worth try to set it up? Coz as I can see from the comments above, Vortex is far from stable.
If it is, kinda, semi-stable, is it possible to integrate Vortex installation into [self-hosted](https://github.com/revoltchat/self-hosted) Revolt?
In any case - PLS share Caddyfile/proxy configs and compose.yaml

**Reply by apeshand** [reply]
Hi.
Yes, you can use vortex for voice communication in the current state. I abandoned the use of `Caddy` in favor of `Traefik` in order not to bother with certificates for domain names.

I also refused to use Minio in any form, replacing it with Yandex S3 (you can use any other compatible S3 service with Amazon S3).

---

I can set aside some time tomorrow and try to post minimal documentation with examples of the configuration files used. Where should I put this?

---

I could also share information about the Revolt desktop client: there you need to replace one string in two files and assemble the client according to the instructions.

**Reply by KamiliaBlow** [reply]
> Hi. Yes, you can use vortex for voice communication in the current state. I abandoned the use of `Caddy` in favor of `Traefik` in order not to bother with certificates for domain names.
> 
> I also refused to use Minio in any form, replacing it with Yandex S3 (you can use any other compatible S3 service with Amazon S3).
> 
> I can set aside some time tomorrow and try to post minimal documentation with examples of the configuration files used. Where should I put this?
> 
> I could also share information about the Revolt desktop client: there you need to replace one string in two files and assemble the client according to the instructions.

do it ❤️

**Reply by phisch** [reply]
I'd love to see some example docker-compose that routes everything through traefik. I have tried setting it up myself, but I'm having issues with the file server and the webrtc voice server.

edit: please ping me here if you found the time to prepare a working example @apeshand 
edit: fileserver works on my end now, only the voice server is still a black-box for me, no idea how the frontend is supposed to communicate with it.

**Reply by apeshand** [reply]
I'm sorry for being late. It took some time to arrange everything in one document.
[Link](https://gist.github.com/apeshand/e663c1dc1bb5ff41e43f00d7c6bb702c "Revolt with voice chat | Windows desktop client") to GitHub Gist.

P.S. ping @phisch :)

**Reply by phisch** [reply]
Thanks for the ping! Already had a bit of a glance, I'll copy your vortex service and adapt it to my traefik setup later. Not sure if I'll get to it today.

**Comment by IAmTomahawkx** [comment]
Just an update on this since I see people are still trying to get Vortex running;
We do not provide support for Vortex, as we consider it legacy software. We are in-progress on rewriting our voice system to use livekit, you can see progress here: https://github.com/revoltchat/backend/pull/318 . If you're having trouble getting Vortex running, consider waiting until this is live. As you guys seem to have figured out, Vortex is kind of an unstable mess.

**Reply by phisch** [reply]
@IAmTomahawkx Are you gonna try implementing screenshare through livekit as well? I believe the zed (editor) team is currently also putting some effort into getting livekit support.

https://github.com/zed-industries/zed/issues/13714
https://github.com/zed-industries/zed/pull/13343

**Reply by IAmTomahawkx** [reply]
Yup that's our current plan. Killing two birds with one stone, as it were.
